src/app/portlets/content-types/fields/content-type-fields-variables/content-type-fields-variables.component.ts

```diff
@@ -52,87 +61,195 @@ export class ContentTypeFieldsVariablesComponent implements OnInit, OnChanges {
+     *
+     * @param {KeyboardEvent} $event
+     * @param {FieldVariable} variable
+     * @param {Number} index
+     * @memberof ContentTypeFieldsVariablesComponent
+     */
+    // tslint:disable-next-line:cyclomatic-complexity
+    onKeyDown($event: KeyboardEvent, fieldVariable: FieldVariable, index: number): void {
+        $event.stopPropagation();
+        if ($event.key === 'Escape') {
+            this.cancelVariableAction(index);
+        } else if ($event.key === 'Enter' && !this.isFieldDisabled(fieldVariable)) {
+            this.saveVariable(fieldVariable, index);
+        }
+    }
+
     /**
      * Handle Delete event
      * @param {FieldVariable} variable
+     * @param {Number} fieldIndex
+     * @memberof ContentTypeFieldsVariablesComponent
+     */
+    deleteVariable(fieldVariable: FieldVariable, fieldIndex: number): void {
+        if (fieldVariable.key !== '') {
+            this.deleteExistingVariable(fieldIndex);
+        } else {
+            this.deleteEmptyVariable(fieldIndex);
+        }
+    }
+
+    /**
+     * Handle Edit button event
+     * @param {Number} fieldIndex
+     * @memberof ContentTypeFieldsVariablesComponent
+     */
+    editVariableAction(fieldIndex: number): void {
+        const key = document.getElementById(`content-type-fields__variables-key-${fieldIndex}`);
+        key.click();
+        this.showEditVariableAction(fieldIndex);
+    }
+
+    /**
+     * Handle Cancel button event
+     * @param {Number} fieldIndex
+     * @memberof ContentTypeFieldsVariablesComponent
+     */
+    cancelVariableAction(fieldIndex: number): void {
+        const editMenu = document.getElementById(`content-type-fields__variables-actions-edit-${fieldIndex}`);
+        const mainMenu = document.getElementById(`content-type-fields__variables-actions-main-${fieldIndex}`);
+        mainMenu.style.display = 'block';
+        editMenu.style.display = 'none';
+
+        if (this.fieldVariablesBackup[fieldIndex].key === '' && this.fieldVariablesBackup[fieldIndex].value === '') {
+            this.deleteVariable(this.fieldVariablesBackup[fieldIndex], fieldIndex);
+        } else {
+            this.fieldVariablesBackup[fieldIndex] = _.cloneDeep(this.fieldVariables[fieldIndex]);
+        }
+        this.table.closeCellEdit();
+    }
+
+    /**
+     * Handle Save event
+     * @param {FieldVariable} variable
+     * @param {Number} variableIndex
+     * @memberof ContentTypeFieldsVariablesComponent
+     */
+    saveVariable(variable: FieldVariable, variableIndex?: number): void {
+        if (typeof variableIndex === 'number') {
+            this.updateExistingVariable(variable, variableIndex);
+        } else {
+            this.addEmptyVariable();
+        }
+    }
+
+    /**
+     * Handle Add new variable button event
+     * @memberof ContentTypeFieldsVariablesComponent
+     */
+    addVariable(): void {
+        this.saveVariable(null);
+        setTimeout(() => {
+            this.editVariableAction(0);
```

Comment:
You can set `0` to be the default value of the param in this method so when  you called with no param it uses `0`